cmd/service-controller/console_server.go

```diff
@@ -72,50 +81,184 @@ type VersionInfo struct {
+			fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s", "NAME", "COUNT", " ", "AGE"))
+			for _, group := range e {
+				fmt.Fprintln(tw, fmt.Sprintf("%s\t%d\t%s\t%s", group.Name, group.Total, " ", time.Since(group.LastOccurrence).Round(time.Second)))
+				for _, detail := range group.Counts {
+					if len(detail.Key) > MaxFieldLength {
+						lines := wrap(detail.Key, MaxFieldLength)
+						log.Printf("wrap(%s) => %v", detail.Key, lines)
+						for i, line := range lines {
+							if i == 0 {
+								fmt.Fprintln(tw, fmt.Sprintf("%s\t%d\t%s\t%s", " ", detail.Count, line, time.Since(detail.LastOccurrence).Round(time.Second)))
+							} else {
+								fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s", " ", " ", line, ""))
+							}
+						}
+					} else {
+						fmt.Fprintln(tw, fmt.Sprintf("%s\t%d\t%s\t%s", " ", detail.Count, detail.Key, time.Since(detail.LastOccurrence).Round(time.Second)))
+					}
+				}
+			}
+			tw.Flush()
 		}
 	})
 }
 
-func (server *ConsoleServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (server *ConsoleServer) serveSites() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		d := server.getData(w)
+		if d != nil {
+			options := r.URL.Query()
+			output := options.Get("output")
+			if output == "json" {
+				bytes, err := json.MarshalIndent(d.Sites, "", "    ")
+				if err != nil {
+					server.httpInternalError(w, fmt.Errorf("Error writing json: %s", err))
+				} else {
+					fmt.Fprintf(w, string(bytes)+"\n")
+				}
+			} else {
+				tw := tabwriter.NewWriter(w, 0, 4, 1, ' ', 0)
+				fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%s\t%s", "ID", "NAME", "EDGE", "VERSION", "NAMESPACE", "URL", "CONNECTED TO"))
+				for _, site := range d.Sites {
+					fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%t\t%s\t%s\t%s\t%s", site.SiteId, site.SiteName, site.Edge, site.Version, site.Namespace, site.Url, strings.Join(site.Connected, " ")))
+				}
+				tw.Flush()
+
+			}
+		}
+	})
+}
+
+func (server *ConsoleServer) serveServices() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		d := server.getData(w)
+		if d != nil {
+			options := r.URL.Query()
+			output := options.Get("output")
+			if output == "json" {
+				bytes, err := json.MarshalIndent(d.Services, "", "    ")
+				if err != nil {
+					server.httpInternalError(w, fmt.Errorf("Error writing json: %s", err))
+				} else {
+					fmt.Fprintf(w, string(bytes)+"\n")
+				}
+			} else {
+				tw := tabwriter.NewWriter(w, 0, 4, 1, ' ', 0)
+				fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s", "ADDRESS", "PROTOCOL", "TARGET"))
+				for _, s := range d.Services {
+					var service *data.Service
+					if hs, ok := s.(data.HttpService); ok {
+						service = &hs.Service
+					}
+					if ts, ok := s.(data.TcpService); ok {
+						service = &ts.Service
+					}
+					if service != nil {
+						fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s", service.Address, service.Protocol, ""))
```

Comment:
Not sure if intended but target is printed on a separate line. 

Also suggest instead of 'target at site name' have a separate column for site to be consistent with other get usage (e.g. would ease parsing)

Reply:
It is intended, as there may be multiple targets for a single service and I thought a line per target would be clearer.

I did wonder about creating a separate column for site. Will do that shortly.

Reply:
@ajssmith I have updated the patch to add the explicit column for site in the services view.